lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```
@@ -485,11 +487,12 @@ addIndices = mapM_ (`rawExecute` [])
 
 delegationToText :: W.WalletDelegation W.PoolId -> Maybe Text
 delegationToText W.NotDelegating = Nothing
-delegationToText (W.Delegating pool) = Just (W.getPoolId pool)
+delegationToText (W.Delegating pool) = Just $ toText pool
 
 delegationFromText :: Maybe Text -> W.WalletDelegation W.PoolId
 delegationFromText Nothing = W.NotDelegating
-delegationFromText (Just pool) = W.Delegating (W.PoolId pool)
+delegationFromText (Just pool) =
+    W.Delegating (fromRight (W.PoolId "invalid id pool") (fromText pool))
```

Hmmm. So here we are silently swallowing a corrupted data. That is not very desirable. Cf my previous comment: https://github.com/input-output-hk/cardano-wallet/pull/748#discussion_r327459984

Having a partial function here is okay-ish, because we don't expect the database to be corrupted and if it is, there's no way to recover from that anyway! Still, I'd prefer defining proper serialization and deserialization of `PoolId` in the `DB.Types` module.

---

added instances in `DB.Types` of `PoolId` - now must fugure out how to use them here :thinking:

---

ok, corrected. Changed to `walDelegation         W.PoolId Maybe     sql=delegation` in TH